keras_cv/models/object_detection/faster_rcnn.py

```diff
@@ -275,21 +280,32 @@ class FasterRCNN(tf.keras.Model):
             (all foreground classes + one background class). By default it uses the rcnn head
             from paper, which is 2 FC layer with 1024 dimension, 1 box regressor and 1
             softmax classifier.
+        prediction_decoder: (Optional)  A `keras.layer` that is responsible for
+            transforming RetinaNet predictions into usable bounding box Tensors.  If
+            not provided, a default is provided.  The default `prediction_decoder`
+            layer uses a `NonMaxSuppression` operation for box pruning.
     """
 
     def __init__(
         self,
         classes,
         bounding_box_format,
+        include_rescaling,
         backbone=None,
-        include_rescaling=False,
         anchor_generator=None,
         rpn_head=None,
         rcnn_head=None,
+        prediction_decoder=None,
         **kwargs,
     ):
+        if bounding_box_format != "yxyx":
```

Comment:
any reason we add this constraint?

Reply:
"yxyx" is hardcoded throughout as the box format. In a follow-up I plan to fix this, but for now I'm just raising an error because if another box format is used this will not behave correctly.